include/caffe/layer.hpp

```diff
@@ -316,6 +317,21 @@ class Layer {
     param_propagate_down_[param_id] = value;
   }
 
+  bool IsForwardPassed() const {
```

Comment:
Is it possible to not have these as a part of `Layer`? AFAICT you can have exactly the same effect by just putting these forward_passed/backward_passed as instance variables in CuDNNBatchNormalizationLayer, and you don't add (essentially) completely unused base functions to a core class. 

IMO it's preferable to keep the core classes like Layer, Blob as small as possible.